### DISCUSSION TITLE
How to change styles for variant outline button only?

### DISCUSSION THREAD

**Discussion by mcmxcdev** [mantinedev/mantine#2004] [Q&A] [answered]
What I have so far is this:

```
  <Button
    styles={(theme) => ({
      root: {
        borderWidth: 2,

        '&:disabled': {
          backgroundColor: theme.colors.red,
          color: theme.colors.green,
        },
      },
    })}
  >
    Foo
  </Button>
```

But in this way, `borderWidth` as well as disabled styling also affects all other variants.

**Comment by rtivital** [comment, marked as answer]
https://codesandbox.io/s/silly-microservice-t11iw1?file=/src/App.tsx

**Reply by VigneshVaidyanathan** [reply]
@rtivital 

Is it possible to do it as part of the theme? This implementation needs this styles to be applied everytime we use an outline button or we have to wrap it under a new component like - `<OutlineButton>` which implements this from the button. Is there a better way to do that instead of creating multiple duplicate components?

The styles API provides classes or styles to change the appearance but it doesnt take the variants, compact other params into consideration.

I am trying to change the styles of the button only when its compact-xs, like instead of 22px height, I want 24px height, everywhere I use.

Is it possible?

**Reply by VigneshVaidyanathan** [reply]
Sorry, just figured it out.

```
<MantineProvider
      theme={{
        components: {
          Button: {
            // Subscribe to theme and component params
            styles: (theme, params: ButtonStylesParams) => ({
              root: {
                padding: '0 30px',
                height: params.compact && params.size === 'xs' ? 24 : undefined,
                backgroundColor:
                  params.variant === 'filled'
                    ? theme.colors[params.color || theme.primaryColor][9]
                    : undefined,
              },
            }),
          },

          Badge: {
            // Use raw styles object if you do not need theme dependency
            styles: {
              root: { borderWidth: 2 },
            },
          },
        },
      }}
    >
      <Button size="xs" compact>Outline button</Button>
      <Button variant="filled" color="cyan">Filled button</Button>
</MantineProvider>
```
